OMCompiler/SimulationRuntime/c/optimization/DataManagement/MoveData.c

```diff
@@ -790,10 +783,13 @@ static inline void setLocalVars(OptData * optData, DATA * data, const double * c
   const int nx = optData->dim.nx;
   const int nv = optData->dim.nv;
 
+
+  memcpy(optData->v[i][j], data->localData[l]->realVars, optData->dim.nReal*sizeof(modelica_real));
```

Comment:
@vruge I think you missed initializing `l` here. Unfortunately, I do not think you can do the `memset(`) here. Entries of `data->localData[:]->realVars` are not contiguous :(.

Reply:
Thanks for looking at it.  I have a lot to do at the moment. I'll look at it around the weekend.

Reply:
NP. Some models are segfaulting on Windows due to this. That's how I found out. 

I can fix it if you tell me what is expected. I was just not sure about the intention.

Reply:
The optimization runs through several times over the time points. With the goal to be improved with each run over the system something. Here all time points in the memory are needed.  I have made a wrong assumption here. Until now I assumed that if I call "functionDAE" with all necessary inputs, that I also get the complete right side of the DAE. But this assumption is wrong, as you can see in the example. The real discrete variables must be copied over from the last run, otherwise they will not be set.

Reply:
or am I wrong? These were the conclusion by the code debugen

Reply:
I have problems with the equation
```
when {$whenCondition1} then
  trapezoid1.T_start = time;
end when;
```
If the variable ` trapezoid1.T_start` has not been set before, it is not defined in functionDAE.